driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -581,9 +583,9 @@ boolean setPool(HostConnectionPool pool) {
      * The connection should generally not be reused after that.
      */
     void release() {
-        HostConnectionPool pool = poolRef.get();
-        if (pool != null)
-            pool.returnConnection(this);
+        Owner owner = ownerRef.get();
+        if (owner instanceof HostConnectionPool)
```

Comment:
:+1:, I'm using your 'MockRequest' implementation and it's working great, thanks!